### PR TITLE
Resolved the deprecation warning in AutoGMM

### LINKED ISSUE #609: [BUG] Deprecation warning in AutoGMM
## Expected Behavior
No warning

## Actual Behavior
Deprecation warning in `AutoGMM`

## Example Code
```
from graspologic.cluster import AutoGMM
```

## Full Traceback
```pytb
FutureWarning: The sklearn.utils.testing module is deprecated in version 0.22 and will be removed in version 0.24. 
The corresponding classes / functions should instead be imported from sklearn.utils. 
Anything that cannot be imported from sklearn.utils is now part of the private API.
```

## Solution 
Caused by https://github.com/microsoft/graspologic/blob/8d7d29f0e569de4b1969d3b1aed8d47168aa6a96/graspologic/cluster/autogmm.py#L14

Just need to replace that decorator with a simple 
```
with warnings.simplefilter('ignore', ConvergenceWarning):
```
(I think)

### PR DESCRIPTION
**Pull request by darkyed**
Fixes #609 

Used python's `warnings` package to suppress `ConvergenceWarnings` and replaced sklearn's `ignore_warnings` decorator

### CODE REVIEW

**Comment by bdpedigo** [comment]
@darkyed thanks for the PR. the mypy test that is failing is not your fault (is currently failing package-wide). i'll review this when I have some time, hopefully later today!

File: graspologic/cluster/autogmm.py [hunk 1]
```diff
@@ -375,87 +376,90 @@ def __init__(
...
+                gm_params["init_params"] = "kmeans"
+            gm_params["reg_covar"] = 0
+            gm_params["max_iter"] = self.max_iter
+
+            criter = np.inf  # if none of the iterations converge, bic/aic is set to inf
+            # below is the regularization scheme
+            while gm_params["reg_covar"] <= 1 and criter == np.inf:
+                model = GaussianMixture(**gm_params)
+                try:
+                    model.fit(X)
```
Comment (bdpedigo) [inline graspologic/cluster/autogmm.py:425]: @darkyed I realized that this line (`model.fit(X)`) is actually the only line where we want to mask the convergence warning - do you think you could refactor so that only this line is in the `warnings.catch_warnings()` context manager?

Comment (darkyed) [reply graspologic/cluster/autogmm.py:423]: @bdpedigo, I have refactored the code as you said. Could you please check now?

**Review by bdpedigo — APPROVED** [review]
looks good to me (all I did was remove changes to the version.txt file which happen automatically if you install in dev mode, we are working on a better way to do this). Thank you for the PR @darkyed! Let us know if there are other issues in the package that you'd like to tackle.
